src/hotspot/share/oops/fieldInfo.cpp

```diff
@@ -113,30 +121,197 @@ Array<u1>* FieldInfoStream::create_FieldInfoStream(GrowableArray<FieldInfo>* fie
   return fis;
 }
 
+int FieldInfoStream::compare_name_and_sig(const Symbol* n1, const Symbol* s1, const Symbol* n2, const Symbol* s2) {
+  int cmp = n1->fast_compare(n2);
+  return cmp != 0 ? cmp : s1->fast_compare(s2);
+}
+
+
+// We use both name and signature during the comparison; while JLS require unique
+// names for fields, JVMS requires only unique name + signature combination.
+typedef struct {
+  Symbol* name;
+  Symbol* signature;
+  int index;
+  int position;
+} field_pos_t;
```

Comment:
The naming of the type and fields is not consistent with HotSpot conventions. Suggestion:

```
struct FieldPosition {
  Symbol* _name;
  Symbol* _signature;
  int _index;
  int _position;
};
```